pdfplumber/table.py

```diff
@@ -59,16 +65,18 @@ def get_group(edge):
         else:
             return ("v", edge["x0"])
 
-    if snap_tolerance > 0:
-        edges = snap_edges(edges, snap_tolerance)
+    if snap_x_tolerance > 0 or snap_y_tolerance > 0:
```

Comment:
Shouldn't it be an `and` comparator? And should <=0 values be treated differently?

Reply:
I don't think so? I think you want to run this block as long as *any* positive snap tolerance is specified. Or perhaps I misunderstand the concern?

Re. `<=0` values:

- `0` should be treated, I think, as indicating "do not snap" and thus no additional code should need to be run.

- I suppose we should convert negative values to `0`. I'll push some code that does that. Unless you see a strong purpose for a negative tolerance here?

Reply:
On negative values, decided instead just to raise an error if they're negative — which gives us more flexibility to change the behavior in the future (rather than silently modifying the user's initial parameters). See https://github.com/jsvine/pdfplumber/pull/553/commits/aa2d594d3b3352dbcef503e4df2e045d69fc2511 below.